---
title: "Tallgrass Restoration Legacies, Summary"
author: "Beau Larkin"
date: "Last updated: `r format(Sys.time(), '%d %B, %Y')`"
output:
  pdf_document:
    df_print: tibble
    fig_caption: yes
    highlight: tango
    toc: true
    toc_depth: 2
geometry: margin = 0.5in
fontsize: 12pt
urlcolor: blue
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, tidy = TRUE, message = FALSE, warning = FALSE, fig.align = 'center')
```

# Packages

```{r libraries, echo = TRUE}
packages_needed = c("tidyverse",
                    "png",
                    "knitr",
                    "conflicted",
                    "formatR")
packages_installed = packages_needed %in% rownames(installed.packages())
if (any(!packages_installed)) {
    install.packages(packages_needed[!packages_installed])
    }
for (i in 1:length(packages_needed)) {
    library(packages_needed[i], character.only = T)
}
conflict_prefer("filter", "dplyr")
conflict_prefer("select", "dplyr")
```

# Introduction

The tallgrass prairie biome in North America is nearly extinct, creating a tremendous desire for restoration or recreation of
this critical habitat. In the upper Midwest, interested landowners have attempted to convert previous agricultural fields
(corn or soybeans, usually) to prairie for decades. The consequences of this practice for soil biota are largely unknown. 
Agricultural fields filter fungal and bacterial communities for species that can tolerate extreme nutrient loads, frequent tillage,
and poor plant diversity. When agricultural practices cease, and native perennial plants are introduced, soil biotic 
communities are again filtered and rapidly change. What is the endpoint of these new communities? Do they begin to resemble 
communities in remnant prairie (the reference condition), or do they form novel communities, indicating a change of state in the 
system due to disturbance associated with agriculture? 

# Description

The goal is to present results and discuss whether a path forward exists. If so, we will determine
the strategy that presents the best story, organization, and interpretation of these results. 

# Data

For this summary, I'll pull as many objects as possible from existing files to reduce the number of 
interspersed code chunks. A few quick new analyses will be necessary, though. Data are loaded here. 

```{r data, echo = TRUE}
sites <- read_csv(paste0(getwd(), "/clean_data/sites.csv"), show_col_types = FALSE) %>%
    mutate(field_type = factor(field_type,ordered = TRUE, levels = c("corn", "restored", "remnant")),
           yr_since = replace(yr_since, which(field_type %in% c("remnant", "corn")), NA)) %>%
    select(-lat, -long, -yr_restore, -yr_rank) %>% 
    arrange(field_key)
```

# Methods

## Sites

The survey followed an unbalanced complete block design. Corn, restored, and remnant fields are compared, with 
at least one of each field type in each block. I have called blocks "regions" so far.
We collected samples and data from four regions (Fig, left).  

```{r fig:site_map,out.width="40%",fig.show="hold",fig.cap="\\label{fig:site_map}(Left) Labels show centroids of regions used for this work. BM = Blue Mounds, FG = Faville Grove, FL = Fermilab, LP = Lake Petite. (Right) Labels show individual fields in the Blue Mounds region."}
include_graphics(c("site_locations_files/figure-gfm/site_map-1.png", "site_locations_files/figure-gfm/site_map_bm-1.png"))
```

## Design

The design is unbalanced because there are more restored fields than corn or remnant. In all but one case,
only single corn and remnant fields were available in each region. This means that we only have
replication to separate field types when using the entire block design.

```{r fields_regions_types_table}
kable(table(sites$region, sites$field_type),
      format = "pandoc",
      caption = "Count of fields by type and region:\nBM = Blue Mounds, FG = Faville Grove,\nFL = Fermilab, LP = Lake Petite")
```

A feature of our design is that restored fields vary in time since restoration, setting up the potential for a chronosequence
in addition to the contrasts with corn or remnant fields. 
The rules for establishing a chronosequence are strict. We cannot call fields from all regions a chronosequence. With 
seven restored fields in a small geographic area, the Blue Mounds fields are our best bet for this (Fig), but we will likely 
have to call this a "pseudochronosequence" and avoid some inferences. Mantel tests (not shown) failed to find correlations 
between soil variables and pairwise distance, giving us a little confidence that we're avoiding systemic bias. 

## Soil Fungal Communities

We collected soil cores from 10 haphazardly-selected locations in each field. Genomic DNA was extracted, and the 
lab and bioinformatics pipeline delivered community data from ITS or 18S regions clustered as OTUs or SVs. In preliminary work,
inferences made with SVs were weaker but not qualitatively different. I continued with OTUs only.  

A few samples had failed to amplify, leaving some fields represented with 9 samples instead of 10. 
This unbalance, which is normally not a problem, became unacceptable for a couple of reasons. First, I had planned to use permutation tests in 
ordinations, and application function `how()` from package `permute` requires balance at the plot (i.e., field) level. Second, 
I wanted to summarize the subsamples at the field level because fields are our replicates. Any comparisons with 
field metadata, where we have one data point per field, would be pseudoreplicated when regressed against subsample level data. 

Also, after rarefaction it was clear that some samples contained very few sequences compared with the 
others. I used an iterative process to remove subsamples with few sequences, choosing a rarefaction cutoff near the plateau of OTU 
recovery at depth (Fig). This ended up being 8 subsamples for ITS and 7 for 18S datasets. These subsample sequence 
values were then rarefied and used as-is, or, when reporting at the field level, subsample sequence values were summed
and rarefied. The resulting data files recovered many more OTUs, but didn't change any major interpretation of the data. The situation with 
18S data is similar, but at a much lower total sequencing depth (not shown). 

```{r fig:its_rcurve_compare,out.width="100%",fig.cap="\\label{fig:its_rcurve_compare}Individual-based rarefaction of ITS OTUs in subsamples when nine subsamples per field are retained (pre, top row) or when eight subsamples are retained (post, bottom row). Vertical lines show the minimum sequencing depth available across samples; this is the depth to which the entire set would be rarefied. Horizontal lines show the number of OTUs that would be recovered from each sample at that rarefaction depth."}
include_graphics("microbial_diagnostics_post_files/its_rarecurve_compare.png")
```

Species (OTU) accumulation was performed on field-level sums of sequences. Most fields were undersampled (Fig), particularly remnant and restored fields 
(ITS) and corn and remnant fields (18S). 

```{r fig:accum,out.width="100%",fig.cap="\\label{fig:accum}Species accumulations for rarefied sums of field sequences for ITS and 18S datasets. Species accumulation by the exact method; standard deviation (vertical lines) conditioned by the empirical dataset."}
include_graphics("microbial_diagnostics_post_files/figure-gfm/rare_species_accum_fig-1.png")
```



